art/defences/preprocessor/thermometer_encoding.py

```diff
@@ -146,6 +146,10 @@ def estimate_gradient(self, x: np.ndarray, grad: np.ndarray) -> np.ndarray:
         :param grad: Gradient value so far.
         :return: The gradient (estimate) of the defence.
         """
+        if self.channels_first:
+            x = np.swapaxes(x, 1, -1)
```

Comment:
If `x` gets modified we should make a copy of it at the beginning and permute the copy or reverse the permutation of `x` in the if block at line 163.